cmd/private-org-peribolos-sync/main.go

```diff
@@ -0,0 +1,254 @@
+	"k8s.io/test-infra/prow/config/secret"
+	"k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/interrupts"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/config"
+	"github.com/openshift/ci-tools/pkg/promotion"
+)
+
+type gitHubClient interface {
+	GetRepo(owner, name string) (github.FullRepo, error)
+}
+
+type stringSlice struct {
+	values []string
+}
+
+func (s *stringSlice) String() string {
+	return strings.Join(s.values, string(filepath.Separator))
+}
+
+func (s *stringSlice) Set(value string) error {
+	s.values = append(s.values, value)
+	return nil
+}
+
+type options struct {
+	peribolosConfig string
+	destOrg         string
+	releaseRepoPath string
+	includeRepos    stringSlice
+	github          flagutil.GitHubOptions
+
+	includeReposSet map[string]sets.String
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	fs.StringVar(&o.peribolosConfig, "peribolos-config", "", "Peribolos configuration file")
+	fs.StringVar(&o.releaseRepoPath, "release-repo-path", "", "Path to a openshift/release repository directory")
+	fs.StringVar(&o.destOrg, "destination-org", "", "Destination name of the peribolos configuration organzation")
+
+	fs.Var(&o.includeRepos, "include-repo", "One or more repositories that do not promote official images and need to be included")
+
+	o.github.AddFlags(fs)
+
+	args := os.Args[1:]
+	if err := fs.Parse(args); err != nil {
+		os.Exit(1)
+	}
+	fs.Parse(args)
+
+	return o
+}
+
+func validateOptions(o *options) error {
+	var validationErrors []error
+
+	if len(o.releaseRepoPath) == 0 {
+		validationErrors = append(validationErrors, errors.New("--release-repo-path is not specified"))
+	}
+	if len(o.peribolosConfig) == 0 {
+		validationErrors = append(validationErrors, errors.New("--peribolos-config is not specified"))
+	}
+	if len(o.destOrg) == 0 {
+		validationErrors = append(validationErrors, errors.New("--destination-org is not specified"))
+	}
+
+	if len(o.includeRepos.values) > 0 {
+		o.includeReposSet = make(map[string]sets.String)
+		for _, repo := range o.includeRepos.values {
+			orgRepo := strings.Split(repo, "/")
+			if len(orgRepo) != 2 {
+				validationErrors = append(validationErrors, fmt.Errorf("--include-repo: %s must be in org/repo format", repo))
+			} else {
+				if _, ok := o.includeReposSet[orgRepo[0]]; !ok {
+					o.includeReposSet[orgRepo[0]] = sets.NewString([]string{orgRepo[1]}...)
+				} else {
+					o.includeReposSet[orgRepo[0]].Insert(orgRepo[1])
+				}
+			}
+		}
+	}
+
+	if err := o.github.Validate(false); err != nil {
+		validationErrors = append(validationErrors, err)
+	}
+
+	return kerrors.NewAggregate(validationErrors)
+}
+
+func main() {
+	o := gatherOptions()
+	err := validateOptions(&o)
+	if err != nil {
+		logrus.WithError(err).Fatal("invalid options")
+	}
+
+	go func() {
+		interrupts.WaitForGracefulShutdown()
+		os.Exit(1)
```

Comment:
?

Reply:
we should exit with non-zero if we catch an interruption signal, right?

Reply:
This is not how you are supposed to use this library -- please see the doc and other callers

Reply:
Thanks, I will check the library out. I was confused because we already use it like this in other tools as well.